LXReorderableCollectionViewFlowLayout/LXReorderableCollectionViewFlowLayout.m

```diff
@@ -138,7 +140,10 @@ - (void)applyLayoutAttributes:(UICollectionViewLayoutAttributes *)layoutAttribut
 }
 
 - (id<LXReorderableCollectionViewDataSource>)dataSource {
-    return (id<LXReorderableCollectionViewDataSource>)self.collectionView.dataSource;
+    if ( _dataSource == nil ) {
+        _dataSource = (id<LXReorderableCollectionViewDataSource>)self.collectionView.dataSource;
+    }
+    return _dataSource;
```

Comment:
I don't see any code that's setting a custom dataSource.  It's just a convenience accessor into `self.collectionView.dataSource`.   Caching this into our own ivar doesn't make sense to me.

If the property is `readonly`, and we declare the implementation of `-(id)dataSource` without referencing `_dataSource`, the ivar won't be created, via auto-synthesis voodoo.